### PR TITLE
Added SSH key support

### LINKED ISSUE #30: Support for SSH based private repositories
Currently it's possible to specify .git-credentials file for cloning private repositories using HTTPS but not SSH. Also if module contain references to another modules using SSH URL terraform failed to initialise workspace.

### What problem are you facing?
I managed to clone terraform module using .git-credentials, but provider failed to initialise workspace because module contains references to another private repos using ssh based url

### How could Crossplane help solve your problem?
Please make git configuration more flexible. Please add support for specifying ssh keys for terraform.

### PR DESCRIPTION
**Pull request by vglafirov**
### Adds ssh keys support 
* Added ssh keys support to connector object
* Creating controller user within image, this is required for ssh
* Adding some logging and debugging messages

-->
Fixes #30

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Tested creating resources using ssh based URL for module and transitive dependencies. 

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -214,14 +237,14 @@ applyArgs, destroyArgs and planArgs will be added to these default arguments.
...
-  directory is not lost. `provider-terraform` __does not persist state__;
+- You must either use remote state or ensure the provider container's `/tf`
+  directory is not lost. `provider-terraform` **does not persist state**;
   consider using the [Kubernetes] remote state backend.
-* If the module takes longer than the supplied `--timeout` to apply the
+- If the module takes longer than the supplied `--timeout` to apply the
   underlying `terraform` process will be killed. You will potentially lose state
   and leak resources.
-* The provider won't emit an event until _after_ it has successfully applied the
+- The provider won't emit an event until _after_ it has successfully applied the
```
Comment (ytsarev) [inline README.md:246]: I think it appeared in the last iteration of this PR as well, there is no reason to modify markdown to some different style which is apparently specific to your IDE :)

**Review by ytsarev — CHANGES_REQUESTED** [review]

**Comment by ytsarev** [comment]
@bobh66 this PR should also fix #54 , thanks to writeable $HOME creation. Do you want to give it a shot for your scenario before the merge?

**Comment by ytsarev** [comment]
@vglafirov do you still want to work on this change?

**Comment by luisdavim** [comment]
any updates on this?

**Comment by ytsarev** [comment]
We just agreed with @vglafirov that me and/or @bobh66 will takeover this PR soon and will try to incorporate it into the next release.

**Comment by davidwincent** [comment]
> any updates on this?

+1 checking the status of this

github seems to have dropped support for password authentication 2021?

```bash
❯ git config --global credential.helper store
❯ git clone https://github.com/***/argocd-apps.git
Cloning into 'argocd-apps'...
Username for 'https://github.com': ***
Password for 'https://***@github.com':
remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed for 'https://github.com/***/argocd-apps.git/'
```

See https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls

**Comment by soony88** [comment]
Would really like this feature to move through, this PR has been sitting here for too long :(

Our organisation enforces SSH authentication so this is a critical feature for us

**Pull request closed without merging**